src/elm/Page/Community.elm

```diff
@@ -402,21 +293,118 @@ viewObjective loggedIn model metadata index objective =
 
 
 
--- VIEW ACTION
+-- UPDATE
+
+
+type alias UpdateResult =
+    UR.UpdateResult Model Msg (External Msg)
 
 
-viewAction : LoggedIn.Model -> Community.Model -> Maybe Posix -> Community.Action -> Html Msg
-viewAction loggedIn metadata maybeDate action =
+type Msg
+    = NoOp
+    | GotTime Posix
+    | CompletedLoadCommunity (Result (Graphql.Http.Error (Maybe Community.Model)) (Maybe Community.Model))
+      -- Objective
+    | ClickedOpenObjective Int
+    | ClickedCloseObjective
+    | GotActionMsg Action.Msg
+
+
+update : Msg -> Model -> LoggedIn.Model -> UpdateResult
+update msg model ({ shared } as loggedIn) =
     let
-        t =
-            loggedIn.shared.translators.t
+        { t } =
+            shared.translators
+    in
+    case msg of
+        NoOp ->
+            UR.init model
+
+        GotTime date ->
+            UR.init { model | date = Just date }
+
+        GotActionMsg (Action.ClaimButtonClicked action) ->
+            model
+                |> UR.init
+                |> UR.addExt
+                    (UpdatedLoggedIn
+                        { loggedIn
+                            | claimingAction =
+                                { status = Action.ConfirmationOpen action
+                                , feedback = Nothing
+                                , needsPinConfirmation = False
+                                }
+                        }
+                    )
```

Comment:
The idea here is to init the claim process inside the global state when the "Claim" button is clicked and let the `LoggedIn` module do the rest.

Reply:
i like this approach